src/connector/with_options.yaml

```diff
@@ -0,0 +1,835 @@
+# THIS FILE IS AUTO_GENERATED. DO NOT EDIT
+
+BigQueryConfig:
+  fields:
+  - name: bigquery.local.path
+    field_type: Option < String >
+    required: false
+  - name: bigquery.s3.path
+    field_type: Option < String >
+    required: false
+  - name: bigquery.project
+    field_type: String
+    required: true
+  - name: bigquery.dataset
+    field_type: String
+    required: true
+  - name: bigquery.table
+    field_type: String
+    required: true
+  - name: s3_credentials
+    field_type: HashMap < String , String >
+    comments: required keys refer to [`crate::aws_utils::AWS_DEFAULT_CONFIG`]
+    required: false
+  - name: r#type
+    field_type: String
+    required: true
+ClickHouseConfig:
+  fields:
+  - name: clickhouse.url
+    field_type: String
+    required: true
+  - name: clickhouse.user
+    field_type: String
+    required: true
+  - name: clickhouse.password
+    field_type: String
+    required: true
+  - name: clickhouse.database
+    field_type: String
+    required: true
+  - name: clickhouse.table
+    field_type: String
+    required: true
+  - name: r#type
+    field_type: String
+    required: true
+DorisConfig:
+  fields:
+  - name: doris.url
+    field_type: String
+    required: true
+  - name: doris.user
+    field_type: String
+    required: true
+  - name: doris.password
+    field_type: String
+    required: true
+  - name: doris.database
+    field_type: String
+    required: true
+  - name: doris.table
+    field_type: String
+    required: true
+  - name: r#type
+    field_type: String
+    required: true
+IcebergConfig:
+  fields:
+  - name: r#type
+    field_type: String
+    required: true
+  - name: force_append_only
+    field_type: bool
+    required: false
+    default: Default::default
+  - name: table.name
+    field_type: String
+    required: true
+  - name: database.name
+    field_type: String
+    required: true
+  - name: catalog.type
+    field_type: Option < String >
+    required: false
+  - name: warehouse.path
+    field_type: String
+    required: true
+  - name: catalog.uri
+    field_type: Option < String >
+    required: false
+  - name: s3.region
+    field_type: Option < String >
+    required: false
+  - name: s3.endpoint
+    field_type: Option < String >
+    required: false
+  - name: s3.access.key
+    field_type: String
+    required: true
+  - name: s3.secret.key
+    field_type: String
+    required: true
+  - name: primary_key
+    field_type: Option < Vec < String > >
+    required: false
+    default: Default::default
+KafkaConfig:
```

Comment:
KafkaConfig is source, while KafkaProperties is sink. This is confusing. Maybe we can add a `type` field, and/or use two files for source and sink